test/autorest/generated/complexgroup/internal/complexgroup/models.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package complexgroup
+
+import "github.com/Azure/azure-sdk-for-go/sdk/azcore"
+
+// Error ...
+type Error struct {
+	Status  string
+	Message string
+}
+
+func newError(resp *azcore.Response) error {
+	err := Error{}
+	if err := resp.UnmarshalAsJSON(&err); err != nil {
+		return err
+	}
+	return err
+}
+
+func (e Error) Error() string {
+	return e.Message
+}
+
+// Basic ..
+type Basic struct {
+	ID    int    `json:"id,omitempty"`
```

Comment:
Fields need to be pointer-to-type, with the exception of maps and slices.  Same applies to `Error` as well (I missed this earlier).

Reply:
You mean we need to have something like this?

// Basic ..
type Basic struct {
	ID    *int       `json:"id,omitempty"`
	Name  *string    `json:"name,omitempty"`
	Color *ColorType `json:"color,omitempty"`
}

Reply:
For now I'm assuming that you do mean what I put above and have pushed the changes. So also done?